test/test_um2netcdf.py

```diff
@@ -740,3 +741,42 @@ def test_fix_level_coord_skipped_if_no_levels(z_sea_rho_data, z_sea_theta_data):
     m_cube = mock.Mock(iris.cube.Cube)
     m_cube.coord.side_effect = iris.exceptions.CoordinateNotFoundError
     um2nc.fix_level_coord(m_cube, z_sea_rho_data, z_sea_theta_data)
+
+
+# 64 to 32 bit data conversion tests
+# NB: skip float63 to float32 overflow as float32 min/max is  -/+ 3.40e+38
+
+def test_convert_32_bit_with_int64(ua_plev_cube):
+    array = np.array([100, 10, 1, 0, -10], dtype=np.int64)
+    ua_plev_cube.data = array
+    um2nc.convert_32_bit(ua_plev_cube)
+    assert ua_plev_cube.data.dtype == np.int32
+
+
+def test_convert_32_bit_overflow_with_int64(ua_plev_cube):
+    array = np.array([3000000000], dtype=np.int64)
+    assert array[0] > np.iinfo(np.int32).max
+    ua_plev_cube.data = array
+
+    with pytest.warns():
+        um2nc.convert_32_bit(ua_plev_cube)
+
+    assert ua_plev_cube.data.dtype == np.int32
```

Comment:
Does the out-of-bounds conversion give a predictable result, and if so, can you compare to ensure that result is what comes out?

Reply:
The under/overflow result should be predictable.

Technically, the under/overflow result is junk data, so I can't think of what is gained by testing the end result. Also, it adds something else to the intent of the test.

One simple solution is updating the under/overflow warnings to recommend the `--64` cmd line option:
https://github.com/ACCESS-NRI/um2nc-standalone/blob/9bd6f270005518060b37d0bb77e4fbf30c4bf672/umpost/um2netcdf.py#L779-L780
That places the onus on the end user. How does that appeal to you?

Reply:
The only reason you'd test the junk data result is to make sure the junk remains the same, i.e., numpy versions haven't changed how it operates. That being said, I think the `--64` cmd line option does the trick.

Reply:
I've fixed adding `--64` in 6c02f7d9.

Reply:
OK, let's see if I can close this out....